ogr/services/gitlab/project.py

```diff
@@ -120,6 +120,15 @@ def _construct_fork_project(self) -> Optional["GitlabProject"]:
             logger.debug(f"Project {self.repo}/{user_login} does not exist: {ex}")
         return None
 
+    def exists(self) -> bool:
+        try:
+            _ = self.gitlab_repo
+            return True
+        except GitlabGetError as ex:
+            if "404 Project Not Found" in str(ex):
```

Comment:
~does it throw 404 for private repos too?~

tested it, it returns 401 and fails with exception as expected... what do you suggest?
- keep the exception
- return `False`, since it's not accessible
- return `True` as exists, but can't be accessed

Reply:
Good point!

I really can't decide...;)

It really returns a different error code for private repos? Is it secure?..;)

I can't decide, maybe `True` since it exists. WDYT?

Reply:
> Does it really return a different error code for private repos? Is it secure?..;)

it does 401 - unauthorized :D i expected 403

> I can't decide, maybe `True` since it exists. WDYT?

i think `True` is ok

Reply:
Fixed.